src/Controller/AccountOverviewController.php

```diff
@@ -55,6 +55,13 @@ public function uploadProfilePicture(Request $request, SerializerInterface $seri
         $user = $this->getUser();
         $userName = $this->getUser()->getUserIdentifier();
         $path = $request->files->get('file');
+
+        $ext = explode(".",$path->getClientOriginalName());//split original filename by dots
+        $ext = $ext[count($ext)-1]; //get last entry (i.e. extension)
+        if (!in_array($ext,["jpg","png","gif"])){ // only allow jpg, png and gif
+            throw new \Exception("invalid file extension! only jpg, png and gif allowed!");
+        }
+
```

Comment:
A small snippet from the Symfony documentation: 
<img alt="image" src="https://github.com/joeel561/OnlyFood/assets/92155191/96f54a8a-bf88-4216-b5f2-b731503c6ca4">
https://symfony.com/doc/5.4/controller/upload_file.html
https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/HttpFoundation/File/File.php#L55

Therefore it would be more advisable to use `guessExtension()` | `getMimeType()` if you want to allow only certain file types.

https://symfony.com/doc/5.4/reference/constraints/File.html#mimetypes

Or even better, use a contraint / validator for it.
https://symfony.com/doc/5.4/reference/constraints/File.html#mimetypes

Reply:
Correct. 
The code would lead to the fact that if someone would like to upload e.g. a file with the name `example.jpg`, that would work without problems, since only on the file extension is checked. 
But if the file would be in reality a HTML file (which could contain Javascript etc.) it would be possible to execute code on foreign devices. 

With the original code it is also possible at the moment, but your changes would unfortunately not prevent this, but only make it slightly more difficult.

So it would make sense not only to rely on the file extension, but also to check the mime type, or to determine the "correct" file type based on it. `guessExtension` would do exactly that, since internally the mime type is evaluated.

A little example how it could look like if the Symfony entity contrainst are used.

```php
// file src/Entity/User.php
...

use Symfony\Component\Validator\Constraints as Assert;

...

/**
 * NOTE: This is not a mapped field of entity metadata, just a simple property.
 * @Vich\UploadableField(mapping="profile_picture", fileNameProperty="profilePictureName", size="imageSize")
 * @Assert\File(
 *     mimeTypes = {"image/*"},
 *     mimeTypesMessage = "Please upload a valid image file"
 * )
 * 
 * @var File|null
 */
private $profilePictureFile;

...
```

This would check on each set of the `$profilePictureFile` if the mime type matches. If it does not match, a validation exception is thrown with the given text.

Or with the `guessExtension` in the provided coded.
```php
...
if (!in_array($path->guessExtension(),["jpg","png","gif"])){ // only allow jpg, png and gif
    throw new \Exception("invalid file extension! only jpg, png and gif allowed!");
}
...
```